wordpress-mobile/WordPress-Android
# Discover tab - error and progress handling

Parent issue #12028

Implements error and progress states on the new Discover tab.


Progress
- if the content is empty, shows fullscreen progress screen
- if the content is not empty and is fetching the first page, shows the Pull-to-refresh progress
- if the content is not empty and is fetching more items, shows progress indicator at the bottom

Error
- if the full screen progress is displayed and an error occurs, show fullscreen error with retry
- if the content is displayed and an error occurs show an error snackbar message

To test:
1. Enable RI FF
2. Clear app data - you might want to slow down your connection on an emulator
3. Open Discover and notice "fullscreen progress is shown
4. Wait for the content to load
5. Pull to refresh and notice "PTR" progress indicator is shown
6. Make sure it disappears when the request finishes
6. Scroll to the bottom and notice "load more progress indicator" is shown
7. Make sure it disappears when the request finishes
8. Scroll to the bottom again and verify the progress indicator still works as expected

-----------------------
1. Enable RI FF
2. Clear app data - you might want to slow down your connection on an emulator
3. Open Discover, wait until the data for discover start loading (not the data for interest screen) and turn on airplane mode
4. Notice "fullscreen discover error is shown"
5. Turn off airplane mode and click on retry
6. Data are loaded
7. Pull to refresh and turn on airplane mode
8. Notice a snackbar an error message is shown and the PTR progress indicator disappears
9. Scroll to the bottom and notice an error message is shown and the load more progress indicator disappears

PR submission checklist:
